File: Math challenge/Challanges/Division.m
//
//  Division.m
//  Math challenge
//
//  Created by Lukas Bischof on 09.01.18.
//  Copyright © 2018 Lukas Bischof. All rights reserved.
//

#import "Division.h"

@implementation Division

- (Challange *)next {
    NSInteger firstNumb = randr(2, 20);
    NSInteger secondNumb = randr(2, 20);
    NSInteger numerator = firstNumb * secondNumb;
    
    NSInteger result, denominator;
    if (randr(0, 1) == 1) {
        result = firstNumb;
        denominator = secondNumb;
    } else {
        result = secondNumb;
        denominator = firstNumb;
    }
    
    return [[Challange alloc] initWithText:SWF(@"%ld / %ld", (long)numerator, (long)denominator) andAnswer:result];
}

- (NSString *)challangeDescription
{
    return @"Division (2-20)";
}

@end

File: Math challenge/challenges/Division.swift
//
// Created by Lukas Bischof on 25.10.19.
// Copyright (c) 2019 Lukas Bischof. All rights reserved.
//

import Foundation

class Division: MathChallengeGenerator {
  let challengeDescription: String = "Division"

  func next() -> Challenge {
    let result = randomInt(2, 20)
    let denominator = randomInt(2, 20)
    let numerator = result * denominator

    return generateChallenge(numerator: numerator, denominator: denominator, result: result)
  }

  private func generateChallenge(numerator: Int, denominator: Int, result: Int) -> Challenge {
    let text = "\(numerator)/\(denominator)"
    let latex = "\(numerator)\\div\(denominator)"

    return Challenge(text: text, latex: latex, andAnswer: result)
  }
}
